src/main/java/io/aiven/kafka/connect/opensearch/DocumentIDStrategy.java

```diff
@@ -45,8 +46,10 @@ public enum DocumentIDStrategy {
     RECORD_KEY("record.key", "Generated from the record's key",
         record -> convertKey(record.keySchema(), record.key()),
         (request, record) ->  {
-            request.versionType(VersionType.EXTERNAL);
-            request.version(record.kafkaOffset());
+            if (!(request instanceof UpdateRequest)) {
```

Comment:
I think we should check for upsert here, not `UpdateRequest`, right?

Reply:
Nope. Data streams are about creation only, not for update

Reply:
well we use Update Request only for upsert record. and it is a workaround since UpdateRequest does not support  versioning

Reply:
> it needs to be done better

let's do it better